libr/anal/p/anal_avr.c

```diff
@@ -2026,35 +2034,256 @@ RAMPX, RAMPY, RAMPZ, RAMPD and EIND:
+			"gpr		eear0	.1		89		0\n"
+			"gpr		eear1	.1		89		0\n"
+			"gpr		eear2	.1		89		0\n"
+			"gpr		eear3	.1		89		0\n"
+			"gpr		eear4	.1		89		0\n"
+			"gpr		eear5	.1		89		0\n"
+			"gpr		eear6	.1		89		0\n"
+			"gpr		eear7	.1		89		0\n"
+
+			"gpr		eearh	.8		89		0\n"
+			"gpr		eear8	.1		89		0\n"
+			"gpr		eear9	.1		89		0\n"
+
+
+			"gpr		eecr	.8		90		0\n"
+			"gpr		eedr	.8		91		0\n"
+			"gpr		eedr0	.1		91		0\n"
+			"gpr		eedr1	.1		91		0\n"
+			"gpr		eedr2	.1		91		0\n"
+			"gpr		eedr3	.1		91		0\n"
+			"gpr		eedr4	.1		91		0\n"
+			"gpr		eedr5	.1		91		0\n"
+			"gpr		eedr6	.1		91		0\n"
+			"gpr		eedr7	.1		91		0\n"
+
+			//TODO: continue from here: https://github.com/vancegroup-mirrors/avr-libc/blob/06cc6ff5e6120b36f1b246871728addee58d3f87/avr-libc/include/avr/iom328p.h#L216
+			//add subregisters?
+
+			"gpr		gtcrr	.8		90		0\n"
+			"gpr		tcnt0	.8		90		0\n"
+			"gpr		ocr0a	.8		90		0\n"
+			"gpr		ocr0b	.8		90		0\n"
+			"gpr		gpior1	.8		90		0\n"
+			"gpr		gpior2	.8		90		0\n"
+			"gpr		spcr	.8		90		0\n"
+			"gpr		spsr	.8		90		0\n"
+			"gpr		spdr	.8		90		0\n"
+			"gpr		acsr	.8		90		0\n"
+			"gpr		smcr	.8		90		0\n"
+			"gpr		mcusr	.8		90		0\n"
+			"gpr		mcucr	.8		90		0\n"
+			"gpr		spmcsr	.8		90		0\n"
+			"gpr		wdtcsr	.8		90		0\n"
+			"gpr		clkpr	.8		90		0\n"
+			"gpr		prr		.8		90		0\n"
+			"gpr		osccal	.8		90		0\n"
+			"gpr		acsr	.8		90		0\n"
+			"gpr		pcicr	.8		90		0\n"
+			"gpr		eicra	.8		90		0\n"
+			"gpr		pcmsk0	.8		90		0\n"
+			"gpr		pcmsk1	.8		90		0\n"
+			"gpr		pcmsk2	.8		90		0\n"
+			"gpr		pcicr	.8		90		0\n"
+			"gpr		timsk0	.8		90		0\n"
+			"gpr		timsk1	.8		90		0\n"
+			"gpr		timsk2	.8		90		0\n"
+			"gpr		pcicr	.8		90		0\n"
+			"gpr		adc		.8		90		0\n"
+			"gpr		adcw	.8		90		0\n"
+			"gpr		adcl	.8		90		0\n"
+			"gpr		adch	.8		90		0\n"
+			"gpr		adcsra	.8		90		0\n"
+			"gpr		adcsrb	.8		90		0\n"
+
+			"gpr		admux	.8		90		0\n"
+			"gpr		didr0	.8		90		0\n"
+			"gpr		didr1	.8		90		0\n"
+			"gpr		tccr1a	.8		90		0\n"
+			"gpr		tccr1b	.8		90		0\n"
+			"gpr		tccr1c	.8		90		0\n"
+			"gpr		tcnt1	.8		90		0\n"
+			"gpr		tcnt1l	.8		90		0\n"
+			"gpr		icr1	.8		90		0\n"
+			"gpr		icr1l	.8		90		0\n"
+			"gpr		icr1h	.8		90		0\n"
+			"gpr		ocr1h	.16		90		0\n"
+			"gpr		ocr1al	.8		90		0\n"
+			"gpr		ocr1ah	.8		90		0\n"
+			"gpr		ocr1al	.8		90		0\n"
+			"gpr		ocr1b	.16		90		0\n"
+			"gpr		ocr1bl	.8		90		0\n"
+			"gpr		ocr1bh	.8		90		0\n"
+			"gpr		tccr2a	.8		90		0\n"
+			"gpr		tccr2b	.8		90		0\n"
+			"gpr		tcnt2	.8		90		0\n"
+			"gpr		ocr2a	.8		90		0\n"
+			"gpr		ocr2b	.8		90		0\n"
+			"gpr		twbr	.8		90		0\n"
+			"gpr		twsr	.8		90		0\n"
+			"gpr		twar	.8		90		0\n"
+			"gpr		twdr	.8		90		0\n"
+			"gpr		twcr	.8		90		0\n"
+			"gpr		twbr	.8		90		0\n"
+			"gpr		twamr	.8		90		0\n"
+			"gpr		ucsr0a	.8		90		0\n"
+			"gpr		ucsr0b	.8		90		0\n"
+			"gpr		ucsr0c	.8		90		0\n"
+			"gpr		ubrr0l	.8		90		0\n"
+			"gpr		ubrr0h	.8		90		0\n"
+			"gpr		udr0	.8		90		0\n"
+			"gpr		ubrr0l	.8		90		0\n"
+			"gpr		ubrr0l	.8		90		0\n"
+			;
+
+		p = realloc(p, strlen(p) + strlen(section_two) + 1);
```

Comment:
use r_strbuf apis, realloc+strlen+strcat are all vulnerable and unrecommended apis to use